### ISSUE TITLE
Taint Analysis: response header values are not a header-injection sink

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1418]
**Describe the bug**

Spun out of #1416's closing note. The `$headers` parameter of `ResponseFactory::make()` and `Illuminate\Http\Response::__construct` is not modelled as a `header` taint sink, so header injection through response headers goes unreported.

```php
// User-controlled team name interpolated into a quoted filename. Psalm reports nothing.
return response()->make($csv, 200, [
    'Content-Type' => 'text/csv',
    'Content-Disposition' => "attachment; filename=\"{$request->input('team')}.csv\"",
]);
```

PHP's `header()` refuses CR/LF, so classic response splitting is blocked at runtime, but the dropped header changes response semantics (Symfony then defaults `Content-Type` to `text/html`, rendering the body), and header values reaching other emitters keep the injection risk.

This is a detection gap (false negative), the mirror image of the false positive class #1416 addressed, and deliberately kept out of PR #1417.

**Impacted Versions**

dev-master after #1348.
